rfc/rfc-92/rfc-92.md

```diff
@@ -0,0 +1,102 @@
+
+## Background
+
+Expanding further, there are plenty of valid technical reasons on why Hudi should allow different storage layouts, under the upper layer reader/writer and table services implementations.
+
+1. We have use-cases, for cloud-native/high performance implementations of timeline (\`HoodieTimeline\`) and metadata (\`HoodieMetadata\` interface). In our use-case, we would like to explore backing them using NoSQL datastore like DynamoDB, for ultra-low latency queries.
+2. Hudi already supports [different](https://github.com/apache/hudi/blob/master/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/storage/HoodieStorageLayout.java) storage formats/layouts. Tables can be bucketed, consistent hashed or organized by having data laid out in order of arrival (default).
+3. Hudi already allows plug-ability and customization at various layers like record merger, indexes and other core write/read paths.
+4. It's very standard practice in databases to allow multiple storage backends (MySQL supports myISAM, innodb/btree, myrocks/lsm). This may be crucial step towards the database northstar vision.
+5. As a long time member of the Hudi community and open-source enthusiast, I think supporting other table formats, even existing ones like Apache Iceberg or Delta Lake, benefits those communities as well.
+    1. For e.g. the Hudi Streamer tool being used at our data lake (and hundreds more) for ingestion/incremental ETL can also benefit other communities.
+    2. Hudi provides out-of-box automatic table management that is manual in projects like Iceberg. With such an implementation, common data lake services can be reused across formats.
+    3. Hudi's high performance writer path can be extended to other formats (to the extent possible, that is not dependent on features like indexing that is only in Hudi's native format)
+    4. there are more such services and functionalities to be unlocked.
+
+
+Some non-technical reasons:
+1. Though Hudi is clearly defined as a platform over the years, there is so much vendor attention in the space for the past couple years, where Hudi is minimized to a table format and compared. This change will help highlight the value of Hudi's open software services, beyond just open formats.
+2. It may be controversial to say this. But, the project has been facing a lot of vendor FUD due to different vendors supporting different table formats. It is neither in the interest nor the business of the project community to be part of vendor wars. Opening up the table format layer to different implementations avoids these distractions for regular OSS contributors with no vendor interests, and helps focus on open-source software design and development.
```

Comment:
First of all. Thanks for airing this so openly. 

Yes. the project has been facing FUD and its been distracting.. 

At-least IMO, the change and spirit of supporting OSS communities needs to come from the big leaders/vendors in the market. Idk if a RFC can change that..

Reply:
Yes its unfortunate that vendors don't consider that normal engineers contribute to the project. I have been part of Hudi community, for almost 6 years now, just using it at the companies I worked for , as a normal user. 

Hudi has about 40-50 monthly active contributors. I just don't want their/our work to be lost in vendor noise. To me, just making more OSS projects work well with each other seems like a step forward, hoping this reduces noise.

Reply:
+1 on avoiding distractions